<?xml version="1.0"?>

<launch>
  <arg name="ns" default="roboscan_nsl3130"/>
  <arg name="nn" default="camera"/>
  <arg name="respawn" default="false"/>  
  <arg name="lens_type" default="2"/>
  <arg name="lens_center_offset_x" default="0"/>
  <arg name="lens_center_offset_y" default="0"/>
  <arg name="image_type" default="2"/>
  <arg name="start_stream" default="true"/>
  <arg name="int0"  default="1000"/>
  <arg name="int1"  default="100"/>
  <arg name="int2"  default="50"/>
  <arg name="int_gray" default="3000"/>
  <arg name="offset" default="0"/>
  <arg name="min_amplitude" default="100"/>
  <arg name="frequence_modulation" default="0"/>
  <arg name="channel" default="0"/>
  <arg name="median_filter" default="false"/>
  <arg name="average_filter" default="false"/>
  <arg name="edge_filter_threshold" default="0"/>
  <arg name="temporal_filter_factor" default="0.3"/>
  <arg name="temporal_filter_threshold" default="300"/>
  <arg name="temporal_edge_filter_threshold_low" default="0"/>
  <arg name="temporal_edge_filter_threshold_high" default="0"/>
  <arg name="interference_detect_limit" default="0"/>
  <arg name="use_last_value" default="false"/>
  <arg name="cartesian" default="true"/>
  <arg name="publish_point_cloud" default="true"/>
  <arg name="transform_angle" default="0"/>
  <arg name="cut_pixels" default="0"/>



  <group ns="$(arg ns)">
    <node pkg="roboscan_nsl3130"
          type="roboscan_publish_node"
          name="$(arg nn)"
          output="screen"
          respawn="$(arg respawn)">

      <param name="lens_type" value="$(arg lens_type)"/>
      <param name="lens_center_offset_x" value="$(arg lens_center_offset_x)"/>
      <param name="lens_center_offset_y" value="$(arg lens_center_offset_y)"/>
      <param name="image_type" value="$(arg image_type)"/>
      <param name="start_stream" value="$(arg start_stream)"/>
      <param name="int0" value="$(arg int0)"/>
      <param name="int1" value="$(arg int1)"/>
      <param name="int2" value="$(arg int2)"/>
      <param name="int_gray" value="$(arg int_gray)"/>
      <param name="offset" value="$(arg offset)"/>
      <param name="min_amplitude" value="$(arg min_amplitude)"/>
      <param name="frequence_modulation" value="$(arg frequence_modulation)"/>
      <param name="channel" value="$(arg channel)"/>
      <param name="median_filter" value="$(arg median_filter)"/>
      <param name="average_filter" value="$(arg average_filter)"/>
      <param name="edge_filter_threshold" value="$(arg edge_filter_threshold)"/>
      <param name="temporal_filter_factor" value="$(arg temporal_filter_factor)"/>
      <param name="temporal_filter_threshold" value="$(arg temporal_filter_threshold)"/>
      <param name="temporal_edge_filter_threshold_low" value="$(arg temporal_edge_filter_threshold_low)"/>
      <param name="temporal_edge_filter_threshold_high" value="$(arg temporal_edge_filter_threshold_high)"/>
      <param name="interference_detect_limit" value="$(arg interference_detect_limit)"/>
      <param name="use_last_value" value="$(arg use_last_value)"/>
      <param name="cartesian" value="$(arg cartesian)"/>
      <param name="publish_point_cloud" value="$(arg publish_point_cloud)"/>

    </node>

    <node name="rviz" pkg="rviz" type="rviz" args= "-d $(find roboscan_nsl3130)/rviz/roboscan_nsl3130rviz.rviz" > </node>

    <node name="rqt_reconfigure" pkg="rqt_reconfigure" type="rqt_reconfigure"> </node>

  </group>
</launch>
